core/templates/services/html-length.service.ts

```diff
@@ -27,74 +27,102 @@ export class HtmlLengthService {
   constructor(
     private loggerService: LoggerService,
     private sanitizer: DomSanitizer
-  ) {}
+  ) { }
 
-  /**
-    * ['oppia-noninteractive-image', 'oppia-noninteractive-math']
-    * The above tags constitutes for non text nodes.
-    */
   nonTextTags = ['oppia-noninteractive-math', 'oppia-noninteractive-image'];
-
-  /**
-    * ['li','blockquote','em','strong', 'p', 'a']
-    * The above tags serve as the tags for text content. Blockquote,
-    * em, a, strong occur as descendants of p tag. li tags are
-    * descendants of ol/ul tags.
-    */
   textTags = ['p', 'ul', 'ol'];
 
   computeHtmlLengthInWords(htmlString: string): number {
     if (!htmlString) {
       this.loggerService.error('Empty string was passed to compute length');
       return 0;
     }
+
     const sanitizedHtml = this.sanitizer.sanitize(
       SecurityContext.HTML, htmlString) as string;
-    let totalWeight = this.calculateBaselineLength(sanitizedHtml);
+    let totalWords = this.calculateBaselineLength(sanitizedHtml);
 
-    // Identify custom tags using regex on the original HTML string.
-    const customTagRegex = /<oppia-noninteractive-(?:math|image)[^>]*>/g;
-    const customTags = htmlString.match(customTagRegex);
+    const customTags = this.extractCustomTags(htmlString);
+    for (const customTag of customTags) {
+      totalWords += this.getWeightForNonTextNodes(customTag, false);
+    }
 
-    if (customTags) {
-      for (const customTag of customTags) {
-        totalWeight += this.getWeightForNonTextNodes(customTag);
-      }
+    return totalWords;
+  }
+
+  computeHtmlLengthInCharacters(htmlString: string): number {
+    if (!htmlString) {
+      this.loggerService.error('Empty string was passed to compute length');
```

Comment:
Let's have the two error messages be different so it's easy to identify where they come from -- e.g. have one mention words and the other characters.